### ISSUE TITLE
Fix Broken Relationships in DB (Source: BETA Test)

### ISSUE THREAD

**Issue by lauratinnel** [ITI/searcch#90]
The database has a number of broken relationships. When a user clicks on the relationship, it doesn't work correctly. Check for these and fix manually.

**Comment by lauratinnel** [comment]
Discussion on Slack 11/1/2021

Laura Tinnel  1:11 PM
When I was cleaning up duplicates, I wound up creating a situation where there was a dangling relationship. I was able to create the correct relationship (to the remaining artifact) and delete the other. I'm just trying to identify any other dangling relationships so I can manually clean them up.

David Johnson  1:14 PM
Due to the way we use foreign keys, this is not possible from a database standpoint.  FWIW, this was exactly the issue with the artifact 393 bug -- we were trying to delete the artifact before fkey refs to it were deleted (ratings/reviews/faves).
This is why I wonder if the bug is in what the back is returning in the relationship list, or in the front's processing of the list.

Laura Tinnel  1:16 PM
Dunno. I guess I can just leave this issue there for now while we figure this out. Maybe it's not such a big problem and eventually, if there is one of these, someone will find it and report it.
1:17
The only way I can think to do this is to write a program that queries all relationships and then looks up the target of the relationship to see if it exists.
New

David Johnson  1:26 PM
That is the query I ran... two left joins (select r.id,r.artifact_id,r.related_artifact_id,al.id,ar.id from artifact_relationships as r left join artifacts as al on r.artifact_id=al.id left join artifacts as ar on r.related_artifact_id=ar.id; -- this would have shown NULL values in either of the final two columns had any dangling refs existed, and it did not, as expected according to the fkey constraint.

Laura Tinnel  1:27 PM
OK
